components/UI/ServiceSubmenu.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import Plus from '../../assets/icons/plus.svg';
+
+const Header = styled.div`
+  display: grid;
+  grid-template-areas:
+    'title toggle'
+    'list list';
+
+  h1 {
+    grid-area: title;
+  }
+`;
+
+const CloseButton = styled.a`
+  grid-area: toggle;
+  align-self: center;
+  justify-self: end;
+`;
+
+const Navigation = styled.ul`
+  grid-area: list;
+  margin-bottom: calc(var(--margin) * 3);
+  list-style: none;
+
+  li {
+    margin-bottom: calc(var(--margin)/2);
+
+    a[disabled] {
+      color: var(--color-me);
+      text-decoration: none;
+      cursor: not-allowed;
+    }
+  }
+`;
+
+export function ServiceSubmenu({ title, children }) {
+    const [opensubmenu, setOpenSubmenu] = useState(false); // @TODO stylelint is throwing an error for line 27 if variable is camelCase. disableing stylelint for the line doesn't work for some reason.
+
+    return (<Header>
+        <h1>{ title }</h1>
+        { React.Children.map(children, child =>
+            React.cloneElement(child, { opensubmenu, setOpenSubmenu }),
+        ) }
+    </Header>
+    );
+}
+
+function Toggle({ opensubmenu, setOpenSubmenu, callback }) {
```

Comment:
No need to componentize `ServiceSubmenu.Toggle` at this point I guess. Or do we plan on having one of those menus *without* the toggle?

When you include the toggle within the `ServiceSubmenu` component right away, then it's easy to keep the state (open/closed) in there. Like this:

```js
   const [menuIsOpen
   return (<Header>
        <h1>{ title }</h1>
        <Toggle ...></Toggle>
        { React.Children.map(children, child =>
            React.cloneElement(child, { opensubmenu, setOpenSubmenu }),
        ) }
    </Header>
    );
```

Reply:
I would argue that over the course of redesigning this might be valuable. Potentially desktop view wouldn't use the toggle.

Reply:
As discussed in person: Desktop vs. mobile does not make a difference for -if- the React component should be a separate component or not. Desktop vs. mobile will be dealt with via CSS.